### PR TITLE
Enumerable

### PR DESCRIPTION
**Pull request by Ashe546**
- initialize
- each class has a separate `.rb` file.
- Created a class MyList that has an instance variable @list.
- Implemented a method `#each` that yields successive members of @list and uses the MyEnumerable module.

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module MyEnumerable
+  # all method
+  def all?
+    array1 = []
+    @list.each { |item| array1.push(item) if yield item }
+    puts array1.length == @list.length
+  end
```
Comment (GzimAsani) [inline enumerable.rb]: - [ ] Good job with your enumerables, but when testing the all method runs more than it should, check the example I tested it with two inputs and it gives the correct answer for example true but then it returns nil as well, kindly update it
image
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @Ashe546 and @MohamedCK                              :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your filter and any method :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 



 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @Ashe546, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
